drivers/can/can_nxp_s32_canxl.c

```diff
@@ -0,0 +1,977 @@
+		data->timing.phase_seg2 = config->phase_seg2;
+		err = can_calc_prescaler(dev, &data->timing, config->bitrate);
+		if (err) {
+			LOG_WRN("Bitrate error: %d", err);
+		}
+	}
+
+	LOG_DBG("Setting CAN bitrate %d:", config->bitrate);
+	nxp_s32_zcan_timming_to_canxl_timming(&data->timing, &config->can_cfg->bitrate);
+
+#ifdef CONFIG_CAN_FD_MODE
+	data->timing_data.sjw = config->sjw_data;
+	if (config->sample_point_data) {
+		err = can_calc_timing_data(dev, &data->timing_data, config->bitrate_data,
+						config->sample_point_data);
+		if (err == -EINVAL) {
+			LOG_ERR("Can't find timing data for given param");
+			return -EIO;
+		}
+		if (err > 0) {
+			LOG_WRN("Sample-point-data err : %d", err);
+		}
+	} else {
+		data->timing_data.prop_seg = config->prop_seg_data;
+		data->timing_data.phase_seg1 = config->phase_seg1_data;
+		data->timing_data.phase_seg2 = config->phase_seg2_data;
+		err = can_calc_prescaler(dev, &data->timing_data, config->bitrate_data);
+		if (err) {
+			LOG_WRN("Bitrate data error: %d", err);
+		}
+	}
+
+	LOG_DBG("Setting CAN-FD bitrate %d:", config->bitrate_data);
+	nxp_s32_zcan_timming_to_canxl_timming(&data->timing_data, &config->can_cfg->Fd_bitrate);
+#endif
+
+	/* Initialize CAN structure*/
+	Canexcel_Ip_Init(config->instance, config->can_cfg, data->can_state);
+
+	/* Enable Interrupt */
+	Canexcel_Ip_EnableInterrupts(config->instance);
+
+	config->irq_config_func();
+
+#ifndef CONFIG_CAN_AUTO_BUS_OFF_RECOVERY
+	config->base_sic->BCFG1 &= (~CANXL_SIC_BCFG1_ABRDIS_MASK);
+#endif
+
+	can_nxp_s32_get_state(dev, &data->state, NULL);
+
+	return 0;
+}
+
+static const struct can_driver_api can_nxp_s32_driver_api = {
+	.get_capabilities = can_nxp_s32_get_capabilities,
+	.start = can_nxp_s32_start,
+	.stop = can_nxp_s32_stop,
+	.set_mode = can_nxp_s32_set_mode,
+	.set_timing = can_nxp_s32_set_timing,
+	.send = can_nxp_s32_send,
+	.add_rx_filter = can_nxp_s32_add_rx_filter,
+	.remove_rx_filter = can_nxp_s32_remove_rx_filter,
+	.get_state = can_nxp_s32_get_state,
+#ifndef CONFIG_CAN_AUTO_BUS_OFF_RECOVERY
+	.recover = can_nxp_s32_recover,
+#endif
+	.set_state_change_callback = can_nxp_s32_set_state_change_callback,
+	.get_core_clock = can_nxp_s32_get_core_clock,
+	.get_max_filters = can_nxp_s32_get_max_filters,
+	.get_max_bitrate = can_nxp_s32_get_max_bitrate,
+	.timing_min = {
+		.sjw = 0x01,
+		.prop_seg = 0x01,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x02,
+		.prescaler = 0x01
+	},
+	.timing_max = {
+		.sjw = 0x04,
+		.prop_seg = 0x08,
+		.phase_seg1 = 0x08,
+		.phase_seg2 = 0x08,
+		.prescaler = 0x100
+	},
+#ifdef CONFIG_CAN_FD_MODE
+	.set_timing_data = can_nxp_s32_set_timing_data,
+	.timing_data_min = {
+		.sjw = 0x01,
+		.prop_seg = 0x01,
+		.phase_seg1 = 0x01,
+		.phase_seg2 = 0x02,
+		.prescaler = 0x01
+	},
+	.timing_data_max = {
+		.sjw = 0x04,
+		.prop_seg = 0x08,
+		.phase_seg1 = 0x08,
+		.phase_seg2 = 0x08,
+		.prescaler = 0x100
+	}
+#endif
+};
+
+#define CAN_NXP_S32_NODE(n)		DT_NODELABEL(can##n)
```

Comment:
You need to use the DT_INST_ macros here, not DT_NODELABEL().

Reply:
This is used like other drivers, supported by NXP S32

Reply:
the HAL is index-based, hence the use of `DT_NODELABEL`. We'll update all existing NXP S32 shim drivers in the future to make use of `DT_INST` macros, for now I'd suggest to keep it as it is for consistency with the rest of the drivers we've added